### ISSUE TITLE
AssertionError invalid chainCode size in Microsoft Edge

### ISSUE THREAD

**Issue by promethe42** [bitpay/bitcore-lib#53]
Hi,

I'm using eth-lightwallet (https://github.com/ConsenSys/eth-lightwallet) to generate accounts in the browser. Everything works great in Firefox (45.0.2, Windows and Ubuntu) and Chrome (49.0.2623, Windows, Ubuntu and Android).

But in Microsoft Edge (25.10586.0.0) I get the following error and I traced it back to bitcore-lib:

```
AssertionError: chainCode has not the expected size: found 64, expected 32
```

The assertion error is raised here:

https://github.com/bitpay/bitcore-lib/blob/b3b18d532ff86855a142688413f60364aec4048a/lib/hdprivatekey.js#L464

My first guess is this buffer is initialized/filled twice in Microsoft Edge for some reason.
Could it be eth-lightwallet misusing bitcore-lib?
Can anyone confirm this part of bitcore-lib works in an app in Microsoft Edge?

How can we fix this?

Thank you !

**Comment by braydonf** [comment]
I would try running https://github.com/bitpay/bitcore-lib/blob/master/test/index.html to see if all the tests pass to confirm.

**Comment by promethe42** [comment]
```
[06:19:46] Error in plugin 'gulp-shell'
Message:
    Command failed: /bin/sh -c ./node_modules/bitcore-build/node_modules/karma/bin/karma start ./node_modules/bitcore-build/karma.conf.js

Details:
    killed: false
    code: 1
    signal: null
    cmd: /bin/sh -c ./node_modules/bitcore-build/node_modules/karma/bin/karma start ./node_modules/bitcore-build/karma.conf.js
    stdout: INFO [karma]: Karma v0.12.37 server started at http://localhost:9876/
INFO [launcher]: Starting browser Firefox
ERROR [launcher]: Cannot start Firefox
        Can not find the binary firefox
        Please set env variable FIREFOX_BIN
```

I'm running the tests in a vagrant VM.
How can just run Karma on `http://localhost:9876/` in my guest VM and then open it in Microsoft Edge on my Windows 10 host ?

Thanks,

**Comment by braydonf** [comment]
Karma may not work well in that case, and why the `test/indexhtml` file exists.

You can setup a basic http server and then view of the local network:
```
cd bitcore-lib
python -m SimpleHTTPServer 8000
```
And then run tests by going to `http://localhost:8000/test/`

**Comment by promethe42** [comment]
Here you go:

https://cocorico.cc/bitcore-lib/test/index.html

Tests pass on:
* IE11
* Firefox 45
* Chrome 49

But do not pass on:
* Edge: 52 failures

Since it works on IE11 I'm guessing it must be something simple that make it fail in Edge.
Any idea?

Thanks,

**Comment by braydonf** [comment]
So [looks like](https://github.com/bitpay/bitcore-lib/issues/55) we just needed to update browserify dependencies. I've updated `bitcore-build` *(that includes browserify dependencies)* and made a new browserify build that passes all tests in [MSEdge 25.10586.0.0](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/linux/). 

The updated build has landed in `0.13.15`

Let us know if there are any other issues. Thanks for the report.
